src/Components/AuthForm.js

```diff
@@ -0,0 +1,139 @@
+import React from "react";
+import "bootstrap/dist/css/bootstrap.min.css";
+import { Button, Col, Form, Row } from "react-bootstrap";
+import { auth } from "./firebase";
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+
+class AuthForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: "",
+      password: "",
+      errorCode: "",
+      errorMessage: "",
+    };
+  }
+
+  setEmail = (e) => {
+    this.setState({ email: e.target.value });
+  };
+  setPassword = (e) => {
+    this.setState({ password: e.target.value });
+  };
```

Comment:
We could probably use the name property on the inputs to decide which key to update. That way we can avoid the duplication here